Dockerfile

```diff
@@ -1,212 +1,62 @@
-  | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
-  )" \
-  && apk add --no-network --virtual .ruby-rundeps $runDeps \
-  bzip2 \
-  ca-certificates \
-  libffi-dev \
-  procps \
-  yaml-dev \
-  zlib-dev \
-  && apk del --no-network .ruby-builddeps \
-  && cd / \
-  && rm -r /usr/src/ruby \
-  # rough smoke test
-  && ruby --version && gem --version && bundle --version
-
-# install things globally, for great justice
-# and don't create ".bundle" in all our apps
-ENV GEM_HOME /usr/local/bundle
-ENV BUNDLE_PATH="$GEM_HOME" \
-  BUNDLE_SILENCE_ROOT_WARNING=1 \
-  BUNDLE_APP_CONFIG="$GEM_HOME"
-# path recommendation: https://github.com/bundler/bundler/pull/6469#issuecomment-383235438
-ENV PATH $GEM_HOME/bin:$BUNDLE_PATH/gems/bin:$PATH
-# adjust permissions of a few directories for running "gem install" as an arbitrary user
-RUN mkdir -p "$GEM_HOME" && chmod 777 "$GEM_HOME"
-# (BUNDLE_PATH = GEM_HOME, no need to mkdir/chown both)
-
-#------------------------------------------------------------------------------------------
-#
-# End of ruby installation
+#------------------------------------------------------------------------------
 #
-#------------------------------------------------------------------------------------------
-
-# Install alpine equivalent for "build-essential"
+# Install requiered bundler version
 #
-# and other dependencies required by dev.to various
-# ruby dependencies (postgresql-dev, tzdata)
-RUN apk add --no-cache alpine-sdk postgresql-dev tzdata
-
-# Im installing bash, as im a bash addict (not that great with sh)
-RUN apk add bash
-
-# Let's setup the rails directory
-# (@TODO - consider a production version?)
-WORKDIR /usr/src/app
-ENV RAILS_ENV development
+#------------------------------------------------------------------------------
+RUN gem install bundler:2.0.2
 
-#####################################################
-#
-# Let's prepare the dev.to source code files
-# WITHOUT docker related files
-#
-# This allow us to modify the docker
-# entrypoint / run file without recompiling
-# the entire application
-# (especially when creating this build script =| )
+#------------------------------------------------------------------------------
 #
-# (@TODO - improve and review ignore to blacklist unneeded items)
+# Install Yarn
 #
-#####################################################
+#------------------------------------------------------------------------------
+ENV PATH=/root/.yarn/bin:$PATH
+RUN apk add --virtual build-yarn curl && \
+    touch ~/.bashrc && \
+    curl -o- -L https://yarnpkg.com/install.sh | sh && \
+    apk del build-yarn
 
+
+#------------------------------------------------------------------------------
 #
-# Prepare the source code and remove any unneeded files
+# Define working directory
 #
-FROM alpine-ruby-node AS source-code-repo
-
-# The workdir
+#------------------------------------------------------------------------------
+RUN mkdir /usr/src/app
```

Comment:
You can omit the mkdir command, as docker will create the working directory itself if it does not already exist. 
```suggestion
```

Reply:
Thanks didn't know that.